src/gwt/src/org/rstudio/studio/client/common/mathjax/display/MathJaxPopupPanel.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * MathJaxPopupPanel.java
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.studio.client.common.mathjax.display;
+
+import org.rstudio.core.client.theme.res.ThemeResources;
+import org.rstudio.core.client.widget.MiniPopupPanel;
+import org.rstudio.core.client.widget.ToolbarButton;
+import org.rstudio.studio.client.common.mathjax.MathJax;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.dom.client.Element;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.resources.client.ClientBundle;
+import com.google.gwt.resources.client.CssResource;
+import com.google.gwt.user.client.ui.FlowPanel;
+import com.google.gwt.user.client.ui.VerticalPanel;
+
+public class MathJaxPopupPanel extends MiniPopupPanel
+{
+   public MathJaxPopupPanel(MathJax mathjax)
+   {
+      super(true, false, false);
+      
+      mathjax_ = mathjax;
+      
+      container_ = new VerticalPanel();
+      contentPanel_ = new FlowPanel();
+      
+      renderToolbar_ = createRenderToolbar();
+      
+      container_.add(contentPanel_);
+      container_.add(renderToolbar_);
+      
+      setWidget(container_);
+      
+      addStyleName(RES.styles().popupPanel());
+   }
+   
+   public void setRenderToolbarVisible(boolean visible)
+   {
+      renderToolbar_.setVisible(visible);
+   }
+   
+   public Element getContentElement()
+   {
+      return contentPanel_.getElement();
+   }
+   
+   private FlowPanel createRenderToolbar()
+   {
+      FlowPanel toolbar = new FlowPanel();
+      
+      initToolbarStyles(toolbar.getElement());
+      
+      ToolbarButton button = new ToolbarButton(
+            ThemeResources.INSTANCE.chevron(),
+            new ClickHandler()
+            {
+               @Override
+               public void onClick(ClickEvent event)
+               {
+                  mathjax_.promotePopupToLineWidget();
+               }
+            });
+      button.setTitle("Promote to Chunk Output");
+      
+      toolbar.add(button);
+      toolbar.setVisible(false);
+      
+      return toolbar;
+   }
+   
+   private static final native void initToolbarStyles(Element el) /*-{
```

Comment:
FWIW there's no need for JSNI here -- you can also use `el.getStyle().setProperty("borderTop", "1px solid #BABABA")`